<launch>
  <arg name="camera_number" default="1"/>
  <arg name="approximate_sync_tolerance_seconds" default="0.0"/>
  <arg name="input/image0" default="image0"/>
  <arg name="input/camera_info0" default="camera_info0"/>
  <arg name="input/image1" default="image1"/>
  <arg name="input/camera_info1" default="camera_info1"/>
  <arg name="input/image2" default="image2"/>
  <arg name="input/camera_info2" default="camera_info2"/>
  <arg name="input/image3" default="image3"/>
  <arg name="input/camera_info3" default="camera_info3"/>
  <arg name="input/image4" default="image4"/>
  <arg name="input/camera_info4" default="camera_info4"/>
  <arg name="input/image5" default="image5"/>
  <arg name="input/camera_info5" default="camera_info5"/>
  <arg name="input/image6" default="image6"/>
  <arg name="input/camera_info6" default="camera_info6"/>
  <arg name="input/image7" default="image7"/>
  <arg name="input/camera_info7" default="camera_info7"/>
  <arg name="input/point_cloud" default="poinc_cloud"/>

  <node_container pkg="rclcpp_components" exec="component_container_mt" name="simple_fusion_container" namespace="">
    <composable_node pkg="simple_fusion" plugin="simple_fusion::SimpleFusionNode" name="simple_fusion">
      <param name="camera_number" value="$(var camera_number)"/>
      <param name="approximate_sync_tolerance_seconds" value="$(var approximate_sync_tolerance_seconds)"/>
      <remap from="~/input/point_cloud" to="$(var input/point_cloud)"/>
      <remap from="~/input/image0" to="$(var input/image0)"/>
      <remap from="~/input/camera_info0" to="$(var input/camera_info0)"/>
      <remap from="~/input/image1" to="$(var input/image1)"/>
      <remap from="~/input/camera_info1" to="$(var input/camera_info1)"/>
      <remap from="~/input/image2" to="$(var input/image2)"/>
      <remap from="~/input/camera_info2" to="$(var input/camera_info2)"/>
      <remap from="~/input/image3" to="$(var input/image3)"/>
      <remap from="~/input/camera_info3" to="$(var input/camera_info3)"/>
      <remap from="~/input/image4" to="$(var input/image4)"/>
      <remap from="~/input/camera_info4" to="$(var input/camera_info4)"/>
      <remap from="~/input/image5" to="$(var input/image5)"/>
      <remap from="~/input/camera_info5" to="$(var input/camera_info5)"/>
      <remap from="~/input/image6" to="$(var input/image6)"/>
      <remap from="~/input/camera_info6" to="$(var input/camera_info6)"/>
      <remap from="~/input/image7" to="$(var input/image7)"/>
      <remap from="~/input/camera_info7" to="$(var input/camera_info7)"/>
      <extra_arg name="use_intra_process_comms" value="true" />
    </composable_node>
  </node_container>
</launch>